src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Parser.cs

```diff
@@ -96,16 +106,36 @@ public partial class RegexGenerator
 
             if (pattern is null || cultureName is null)
             {
-                return new DiagnosticData(DiagnosticDescriptors.InvalidRegexArguments, GetComparableLocation(methodSyntax), "(null)");
+                return new DiagnosticData(DiagnosticDescriptors.InvalidRegexArguments, GetComparableLocation(memberSyntax), "(null)");
             }
 
-            if (!regexMethodSymbol.IsPartialDefinition ||
-                regexMethodSymbol.IsAbstract ||
-                regexMethodSymbol.Parameters.Length != 0 ||
-                regexMethodSymbol.Arity != 0 ||
-                !SymbolEqualityComparer.Default.Equals(regexMethodSymbol.ReturnType, regexSymbol))
+            bool nullableRegex;
+            if (regexMemberSymbol is IMethodSymbol regexMethodSymbol)
             {
-                return new DiagnosticData(DiagnosticDescriptors.RegexMethodMustHaveValidSignature, GetComparableLocation(methodSyntax));
+                if (!regexMethodSymbol.IsPartialDefinition ||
+                    regexMethodSymbol.IsAbstract ||
+                    regexMethodSymbol.Parameters.Length != 0 ||
+                    regexMethodSymbol.Arity != 0 ||
+                    !SymbolEqualityComparer.Default.Equals(regexMethodSymbol.ReturnType, regexSymbol))
+                {
+                    return new DiagnosticData(DiagnosticDescriptors.RegexMemberMustHaveValidSignature, GetComparableLocation(memberSyntax));
+                }
+
+                nullableRegex = regexMethodSymbol.ReturnNullableAnnotation == NullableAnnotation.Annotated;
+            }
+            else
+            {
+                Debug.Assert(regexMemberSymbol is IPropertySymbol);
+                IPropertySymbol regexPropertySymbol = (IPropertySymbol)regexMemberSymbol;
+                if (!memberSyntax.Modifiers.Any(SyntaxKind.PartialKeyword) || // TODO: Switch to using regexPropertySymbol.IsPartialDefinition when available
+                    regexPropertySymbol.IsAbstract ||
+                    regexPropertySymbol.SetMethod is not null ||
+                    !SymbolEqualityComparer.Default.Equals(regexPropertySymbol.Type, regexSymbol))
+                {
+                    return new DiagnosticData(DiagnosticDescriptors.RegexMemberMustHaveValidSignature, GetComparableLocation(memberSyntax));
+                }
+
+                nullableRegex = regexPropertySymbol.NullableAnnotation == NullableAnnotation.Annotated;
```

Comment:
The value of `IPropertySymbol.NullableAnnotation` value does not take into account annotations deriving from attributes (`NotNullAttribute`, `MaybeNullAttribute`, etc) which need to be checked manually. We have helper code that does this in the STJ generator:

https://github.com/dotnet/runtime/blob/59e8bbcf83b664c3de6cfa553d9bbfad76578765/src/libraries/System.Text.Json/gen/Helpers/RoslynExtensions.cs#L301-L383

but adding a suppression at the generated expression is the quickest way to get guaranteed results.

Reply:
That should not be necessary or desirable here. The only reason the nullable information is relevant is for the generated partial to match the manually-written partial. For annotations, that's relevant, or else the compiler may warn that they differ. But for attributes, we actually wouldn't want to duplicate them, as all attributes from across all partial declarations are accumulated, and if we duplicated them, we'd get compilation failures due to multiples not being allowed. I've augmented the tests to confirm that.